### DISCUSSION TITLE
Is it possible to run part of the script in headless mode and then show the browser to run the rest?

### DISCUSSION THREAD

**Discussion by LintaoAmons** [getgauge/taiko#2682] [General]
because headless mode is considerably faster, but still want to interact with the browser after taiko have help me to finish the tedious work~

**Comment by Nitjs3fni** [comment]
Hi, I think what you would like to do can be achieved with hooks ([documentation](https://docs.gauge.org/writing-specifications.html?os=windows&language=javascript&ide=vscode#execution-hooks-1)) and two tagged scenarios. 

For hooks part:
Add two beforeScenarios with proper tags to your setup file.
```
beforeScenario(async () => {
	await openBrowser({headless: true});
}, {tags: ["headless"]});
```
```
beforeScenario(async () => {
	await openBrowser({headless: false});
}, {tags: ["headful"]});
```
```
afterScenario(async () => {
	await closeBrowser();
}); 
(this is necessary as you can't open another browser instance when one is already running within scope of one spec)
```

For Scenarios part:
In a spec add two scenarios or create two specs (first with Scenario A and second with Scenario B for parallel execution)
```
## Scenario A 
Tags: headless
* /your steps/
(this scenario will be executed in headless mode)
```
```
## Scenario B 
Tags: headful
* /your steps/
(this scenario will be executed in headful mode)
```
